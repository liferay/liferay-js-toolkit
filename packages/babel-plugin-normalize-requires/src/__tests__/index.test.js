import * as babel from 'babel-core';
import plugin from '../index';

describe('when requiring local modules', () => {
	it('keeps trailing ".js" from package names', () => {
		const source = `
	    require('a-package.js')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('keeps trailing ".js" from package names', () => {
		const source = `
	    require('@some-scope/a-package.js')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('removes trailing ".js" from module names', () => {
		const source = `
	    require('./a-module.js')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('removes trailing "/" from module names', () => {
		const source = `
		require('./a-module/')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('removes trailing "/" from module names ending in ".js"', () => {
		const source = `
		require('./a-module.js/')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});
});

describe('when requiring external modules', () => {
	it('removes trailing ".js" from module names', () => {
		const source = `
		require('a-package/a-module.js')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('removes trailing "/" from module names', () => {
		const source = `
		require('a-package/a-module/')
	    `;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('removes trailing "/" from module names ending in ".js"', () => {
		const source = `
		require('a-package/a-module.js/')
		`;

		const { code } = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});
});
