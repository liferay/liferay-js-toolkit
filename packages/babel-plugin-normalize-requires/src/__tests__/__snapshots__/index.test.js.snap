// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when requiring external modules removes trailing ".js" from module names 1`] = `
"
require('a-package/a-module');"
`;

exports[`when requiring external modules removes trailing "/" from module names 1`] = `
"
require('a-package/a-module');"
`;

exports[`when requiring external modules removes trailing "/" from module names ending in ".js" 1`] = `
"
require('a-package/a-module.js');"
`;

exports[`when requiring local modules keeps trailing ".js" from package names 1`] = `
"
require('a-package.js');"
`;

exports[`when requiring local modules keeps trailing ".js" from package names 2`] = `
"
require('@some-scope/a-package.js');"
`;

exports[`when requiring local modules removes trailing ".js" from module names 1`] = `
"
require('./a-module');"
`;

exports[`when requiring local modules removes trailing "/" from module names 1`] = `
"
require('./a-module');"
`;

exports[`when requiring local modules removes trailing "/" from module names ending in ".js" 1`] = `
"
require('./a-module.js');"
`;
